website/docs/configuration/resources.html.md

```diff
@@ -70,24 +96,28 @@ and apply across all resource types. (See [Meta-Arguments](#meta-arguments) belo
 
 ### Documentation for Resource Types
 
-[Terraform's provider documentation][providers] is the primary place to
-learn which resource types are available and which arguments to use for each
-resource type. Once you understand Terraform's basic syntax, the provider
-documentation will be where you spend the majority of your time on this website.
+Every Terraform provider has its own documentation, describing its resource
+types and their arguments.
 
-The "[Providers][]" link at the top level of the navigation sidebar will take
-you to an alphabetical list of all of the providers distributed by HashiCorp.
-You can find a specific provider in this master list, or choose a category from
-the navigation sidebar to browse a more focused list of providers.
+Most publicly available providers are distributed on the
+[Terraform Registry](https://registry.terraform.io/browse/providers), which also
+hosts their documentation. When viewing a provider's page on the Terraform
+Registry, you can click the "Documentation" link in the header to browse its
+documentation. Provider documentation on the registry is versioned, and you can
+use the dropdown version menu in the header to switch which version's
+documentation you are viewing.
 
-You can also search GitHub or other sources for third-party providers, which can
-be installed as plugins to enable an even broader selection of resource types.
+To browse the publicly available providers and their documentation, see
+[the providers section of the Terraform Registry](https://registry.terraform.io/browse/providers).
 
-[providers]: /docs/providers/index.html
+-> **Note:** Provider documentation used to be hosted directly on terraform.io,
+as part of Terraform's core documentation. Although some provider documentation
+might still be hosted here, the Terraform Registry is now the main home for all
+public provider docs.
```

Comment:
I'm not sure if this exception is significant enough to warrant mention right here, but unfortunately the special built-in provider `terraform.io/builtin/terraform` (where `terraform_remote_state` lives) doesn't have a presence in the registry because it's not independently-installable, and so we haven't yet made a plan to move its documentation to anywhere other than `terraform.io`. :thinking:

Reply:
Oh!!! Ha, that's an interestingly sticky one. 

I almost feel like we should re-purpose the registry page for the archived hashicorp/terraform provider ([here](https://registry.terraform.io/providers/hashicorp/terraform/latest/docs)) as the "real" docs for the built-in terraform provider, just for consistency. ...Or is that a bad idea?

Reply:
Well, I'll note the exception for the time being, and maybe raise the re-purposing question elsewhere.

Reply:
I expect that would require a fair number of special cases in the Terraform Registry codebase to ensure it appears as `terraform.io/builtin/terraform`, to allow it to pick up the docs out of a funny location the `hashicorp/terraform` repository instead of the usual convention, to show it as belonging to Terraform's own release versions rather than having its own release cadence, and probably several other things.

It is pretty odd to have the old separately-published provider listed in the registry though, because AFAIK the latest release of it isn't even Terraform v0.12/v0.13 compatible so there's no situation where it would make sense to list `source = "hashicorp/terraform"`. I expect there is something we could do to help make this more discoverable with _fewer_ weird special cases in Terraform Registry, but I think that's a tricky enough thing that we should keep it for another day. :grinning: